src/devices/xt300.c

```diff
@@ -0,0 +1,85 @@
+/* XH300 / XT300 Soil Moisture Sensor
+ *
+ * https://www.plantcaretools.com/product/wireless-moisture-monitor/
+ *
+ * Data is transmitted with 6 bytes row:
+ * 
+ +.  0. 1. 2. 3. 4. 5
+ *  FF ID SM TT ?? CC
+ * 
+ * FF: initial preamble
+ * ID: 0101 01ID
+ * SM: soil moisure (decimal 05 -> 99 %)
+ * TT: temperature °C + 40°C (decimal)
+ * ??: always FF... maybe spare bytes
+ * CC: check sum (simple sum) except 0xFF preamble
+  * 
+ */
+
+#include "decoder.h"
+
+static int xt300_callback(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    int ret = 0;
+    int row;
+    int chk;
+    uint8_t *b;
+    int channel, temp;
+    float temp_c;
+    int moisture;
+    data_t *data;
+
+    for (row = 0; row < bitbuffer->num_rows; row++) {
+        if (bitbuffer->bits_per_row[row] != 48) {
+            continue;
+        }
+        b = bitbuffer->bb[row];
+        if (b[0] != 0xFF && b[1] | 0x1 == 0x55) {
+            continue;
+        }
+        chk = add_bytes(b + 1, 4); // sum bytes 1-4
+        chk = chk & 0xFF;
+        if (chk != 0 & chk != b[5] ) {
+            continue;
+        }
+
+        channel  = (b[1] & 0x03);
+        temp     = b[3] - 40;
+        temp_c    = temp * 1.;
+        moisture =  b[2];
+
+        data = data_make(
+            "model",            "",             DATA_STRING, "XT300 Temperature & Moisture",
+            "channel",          "Channel",      DATA_INT,    channel,
+            "temperature_C",    "Temperature",  DATA_FORMAT, "%.01f C", DATA_DOUBLE, temp_c,
+            "moisture",         "Moisture",     DATA_FORMAT, "%d %%", DATA_INT, moisture,
+            "mic",              "Integrity",    DATA_STRING, "CHECKSUM",
```

Comment:
why is there a field that's hardcoded to CHECKSUM?

Reply:
That's a marker to show there is a valid checksum

Reply:
ah okay, that's useful